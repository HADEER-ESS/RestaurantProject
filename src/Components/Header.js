import React from 'react';
import MaterialIcons from 'react-native-vector-icons/dist/MaterialIcons';
import {
  BackHandler,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

const Header = () => {
  return (
    <View style={styles.HeaderStyle}>
      <TouchableOpacity
        style={styles.IconBackground}
        onPress={() => BackHandler.exitApp()}>
        <MaterialIcons name="arrow-back-ios" color="#ffffff" size={19} />
      </TouchableOpacity>
      <Text style={styles.TextStyle}>Restaurants</Text>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  HeaderStyle: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 17,
    // justifyContent: 'space-around',
  },
  IconBackground: {
    backgroundColor: '#000',
    padding: 9,
    borderRadius: 7,
  },
  TextStyle: {
    color: '#000000',
    fontSize: 25,
    fontWeight: '700',
    paddingLeft: '25%',
  },
});
