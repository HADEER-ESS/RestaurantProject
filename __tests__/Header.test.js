import React from 'react';
import Header from '../src/Components/Header';
import renderer from 'react-test-renderer';

it('Test If The Component Render Correctly Across Screen', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
