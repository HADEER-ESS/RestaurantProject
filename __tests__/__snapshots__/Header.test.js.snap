// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test If The Component Render Correctly Across Screen 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 17,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#000",
        "borderRadius": 7,
        "opacity": 1,
        "padding": 9,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 19,
          },
          undefined,
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î— 
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 25,
        "fontWeight": "700",
        "paddingLeft": "25%",
      }
    }
  >
    Restaurants
  </Text>
</View>
`;
